example/parts/product.go

```diff
@@ -3,20 +3,37 @@ package parts
 import (
 	"net/http"
 
+	ct "github.com/theplant/containers"
 	"github.com/theplant/containers/example/models"
 	"github.com/theplant/containers/example/parts/templates"
 )
 
-func Product(r *http.Request) (html string, err error) {
-	ctx := r.Context()
+type Product struct {
+	ProductImages ct.Container
+	ProductColors ct.Container
+}
 
-	p := &models.Product{Name: "iPhone 7"}
+func (prod *Product) Render(r *http.Request) (html string, err error) {
+	// ctx := r.Context()
 
-	if addToCardError := ctx.Value("events.AddToCart.error"); addToCardError != nil {
-		html = addToCardError.(error).Error()
-		return
+	var productImagesHtml, productColorsHtml string
+
+	if prod.ProductImages != nil {
+		productImagesHtml, err = prod.ProductImages.Render(r)
```

Comment:
if add a `Children() []Container` interface, But the parent container needs to have a reference to Render it to some specific place inside it's html.

Can't think a way to let this new added interface both used by reloading to get specific containers. AND let the parent container easily reference one of them and embed it into parent's html.